lib/modules/backupAndRestore/providers/GoogleDrive.js

```diff
@@ -28,9 +34,46 @@ export class GoogleDrive extends Provider {
 			await Permissions.request(['https://accounts.google.com/o/oauth2/v2/auth', 'https://content.googleapis.com/drive/v3/*']);
 		});
 
+		// Re-check email address cache, in case it got invalidated due to google account logout
+		if (!this.getCachedEmailAddress()) {
+			try {
+				this.emailAddress = await this.verifyToken();
+				this.setCachedEmailAddress(this.emailAddress);
+			} catch (err) {
+				console.error(err);
+			}
+		}
+
 		return this;
 	}
 
+	getCachedEmailAddress() {
+		const cachedEmailAddress = localStorage.getItem(CACHED_GOOGLE_EMAIL_ADDRESS);
+		if (cachedEmailAddress) {
+			return cachedEmailAddress;
+		}
+
+		return '';
```

Comment:
These lines could just be `return cachedEmailAddress || '';`

Reply:
Thanks, nice catch. Had missed that while thinking about the JSON parsing.